apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -79,3 +217,292 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
+    }
+
+    def _if_eager(x: Iterator[RT]) -> Iterator[RT]:
+        return EagerIterator(x, pool=pool) if eager else x
+
+    # blockwise reader
+    def _table_reader() -> SingleAxisTableIterReturn:
+        kwargs: Dict[str, object] = {"result_order": sr.result_order}
+        for coord_chunk in _coords_strider(coords[axis], sr.shape[axis], size):
+            sr.reset(**kwargs)
+            step_coords = [*coords]
+            step_coords[axis] = coord_chunk
+            array._set_reader_coords(sr, step_coords)
+            yield pa.concat_tables(_arrow_table_reader(sr)), tuple(
+                joinids[0:axis] + [pa.array(coord_chunk)] + joinids[axis + 1 : ndim]
+            )
+
+    # with reindexing
+    def _reindexed_table_reader() -> SingleAxisTableIterReturn:
+        _tbl_iter = _table_reader() if eager else _if_eager(_table_reader())
+        for tbl, coords in _tbl_iter:
+            pytbl = {}
+            for d in range(ndim):
+                col = tbl.column(f"soma_dim_{d}")
+                if d in axes_to_reindex:
+                    if d == axis:
+                        col = pd.Index(coords[axis]).get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                    else:
+                        col = minor_axes_indexer[d].get_indexer(col.to_numpy())  # type: ignore[no-untyped-call]
+                pytbl[f"soma_dim_{d}"] = col
+            pytbl["soma_data"] = tbl.column("soma_data")
+            yield pa.Table.from_pydict(pytbl), coords
+
+    if eager and not pool:
+        with concurrent.futures.ThreadPoolExecutor() as _pool:
+            pool = _pool
+            yield from _reindexed_table_reader() if len(
+                axes_to_reindex
+            ) else _table_reader()
+    else:
+        yield from _reindexed_table_reader() if len(
+            axes_to_reindex
+        ) else _table_reader()
+
+
+def _scipy_sparse_iter(
+    array: "SparseNDArray",
+    sr: clib.SOMAArray,
+    coords: options.SparseNDCoords,
+    axis: int,
+    size: int,
+    compress: bool,
+    reindex_disable: List[int],
+    eager: bool,
+    pool: Optional[concurrent.futures.Executor] = None,
+) -> Iterator[BlockwiseScipyReadIterResult]:
+    """
+    Private. Iterator over SparseNDArray producing sequence of scipy sparse matrix.
+    """
+
+    IndicesType = Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]]
+    IJDType = Tuple[
+        Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]], npt.NDArray[np.generic]
+    ]
+
+    if len(sr.shape) != 2 or len(coords) > 2 or axis not in [0, 1]:
+        raise SOMAError("SciPy sparse matrix iterator compatible only with 2D arrays")
+
+    if axis in reindex_disable:
+        raise SOMAError(
+            "Unable to disable reindexing of coordinates on CSC/CSR major axis"
+        )
+
+    shape = cast(Tuple[int, int], tuple(sr.shape))
+
+    # normalize params tuple
+    coords = tuple(coords[i] if i < len(coords) else None for i in range(2))
```

Comment:
Since this appears in a couple places, it might be worth pulling out into a little function, largely so it’s easier to understand at a glance (maybe `_pad_with_none(coords, 2)`)